src/util/pelz_io.c

```diff
@@ -201,140 +202,378 @@ int write_to_pipe(char *msg)
+
+  if (!msg)
+  {
+    pelz_log(LOG_ERR, "Unable to allocate memory.");
+    return (1);
+  }
+  memcpy(msg, message, message_length);
+  msg[msg_len - 1] = '\0';
+
+  size_t token_count = 1;
+
+  for (size_t i = 0; i < msg_len; i++)
+  {
+    if (msg[i] == ' ')
+    {
+      token_count++;
+    }
+  }
+  *num_tokens = token_count;
+  char **ret_tokens = (char **) malloc(token_count * sizeof(char *));
+
+  if (!ret_tokens)
+  {
+    pelz_log(LOG_ERR, "Unable to allocate memory.");
+    free(msg);
+    return (1);
+  }
+  char *save = msg;
+  char *token = strtok(msg, " ");
+
+  ret_tokens[0] = (char *) malloc(strlen(token) * sizeof(char) + 1);
+  if (!ret_tokens[0])
+  {
+    pelz_log(LOG_ERR, "Unable to allocate memory.");
+    free(save);
+    return (1);
+  }
+  memcpy(ret_tokens[0], token, strlen(token) + 1);  //copy the '\0'
+  for (size_t i = 1; i < token_count; i++)
+  {
+    char *token = strtok(NULL, " ");
+
+    ret_tokens[i] = (char *) malloc(strlen(token) * sizeof(char) + 1);
+    if (!ret_tokens[i])
+    {
+      pelz_log(LOG_ERR, "Unable to allocate memory.");
+      for (size_t j = 0; j < i; j++)
+      {
+        free(ret_tokens[j]);
+      }
+      free(ret_tokens);
+      free(save);
+      return (1);
+    }
+    memcpy(ret_tokens[i], token, strlen(token) + 1);  //copy the '\0'
+  }
+  if (strtok(NULL, " ") != NULL)
+  {
+    pelz_log(LOG_ERR, "Unable to tokenize pipe message: %s", msg);
+    for (size_t i = 0; i < token_count; i++)
+    {
+      free(ret_tokens[i]);
+    }
+    free(ret_tokens);
+    free(save);
+    return (1);
+  }
+  free(save);
+  *tokens = ret_tokens;
+  return (0);
+}
+
+int parse_pipe_message(char **tokens, char **response)
+{
+  int ret;
+  char *path_ext = NULL;
   charbuf key_id;
-  charbuf path;
+  uint8_t *nkl_data = NULL;
+  size_t nkl_data_len = 0;
+  char *authString = NULL;
+  size_t auth_string_len = 0;
+  const char *ownerAuthPasswd = "";
+  size_t oa_passwd_len = 0;
+  uint8_t *data = NULL;
+  size_t data_length = 0;
+  uint64_t handle;
 
 /*
- *  -e    exit     Terminate running pelz-service
- *  -l    load     Loads a value of type <type> (currently either cert or private)
- *  -c    cert     Server certificate
- *  -p    private  Private key for connections to key servers
- *  -r    remove   Removes a value of type <target> (currently either cert or key)
- *  -k    key      Key with a specified id
- *  -a    all      Indicate all key or cert
+ *  -1    exit              Terminate running pelz-service
+ *  -2    load cert         Loads a server certificate
+ *  -3    load private      Loads a private key for connections to key servers
+ *  -4    remove cert       Removes a server certificate    
+ *  -5    remove all certs  Removes all server certificates
+ *  -6    remove key        Removes a key with a specified id
+ *  -7    remove all keys   Removes all keys
  */
-
-  if (memcmp(msg, "pelz -", 6) == 0)
+  if (memcmp(tokens[0], "pelz", 4) == 0)
```

Comment:
I don't know that we need to check ```tokens[0] == "pelz"```. By not making this check, if another program ever needs to write to the pipe, it should "just work" if they follow the API correctly. They could then use this to send their own program name. I think the only thing we want to do with tokens[0] is use it to create more useful return messages.